lib/testing/testDataCheck.hpp

```diff
@@ -24,6 +25,98 @@
 #include <type_traits> // for is_same, enable_if
 #include <vector>      // for vector
 
+#define CHECK_META_SCALAR_INT_DIRECT(FIELD, META1, META2, ERR_COUNT, BUF_NAME1, FRAME_ID1,         \
+                                     BUF_NAME2, FRAME_ID2)                                         \
+    do {                                                                                           \
+        INFO("Checking meta field {:s}", #FIELD);                                                  \
+        if ((META1)->FIELD != (META2)->FIELD) {                                                    \
+            ERROR("metadata {:s}[{:d}] {:s} != {:s}[{:d}] {:s}; values: {:d} {:d}", (BUF_NAME1),   \
```

Comment:
Will we ever want _NON_OO versions of these?
Maybe we could let the user specify if they want this called at the level of ERROR, FATAL_ERROR, WARN.